swarm-cluster-using-docker-extension/azuredeploy.json

```diff
@@ -0,0 +1,396 @@
+		"subnetRef": "[concat(variables('vnetID'),'/subnets/',variables('subnetName'))]",
+		"vmName": "swarmVM",
+		"vmSize": "Standard_D1",
+		"imagePublisher": "Canonical",
+		"imageOffer": "UbuntuServer",
+		"OSDiskName": "osdiskforlinuxsimple",
+		"vmStorageAccountContainerName": "vhds",
+            	"nsgName" : "swarmNSG",
+            	"nsgID": "[resourceId('Microsoft.Network/networkSecurityGroups',variables('nsgName'))]"
+    },
+    "resources": [
+    	{
+		"type": "Microsoft.Storage/storageAccounts",
+		"name": "[parameters('newStorageAccountName')]",
+		"apiVersion": "2015-05-01-preview",
+		"location": "[variables('location')]",
+		"properties": {
+			"accountType": "[variables('storageAccountType')]"
+		}
+	},
+	{
+		"apiVersion": "2015-05-01-preview",
+		"type": "Microsoft.Network/publicIPAddresses",
+		"name": "[concat(variables('publicIPAddressName'), 'master')]",
+		"location": "[variables('location')]",
+		"properties": {
+			"publicIPAllocationMethod": "[variables('publicIPAddressType')]",
+			"dnsSettings": {
+				"domainNameLabel": "[concat(parameters('dnsNameForPublicIP'), 'master')]"
+			}
+		}
+	},
+	{
+		"apiVersion": "2015-05-01-preview",
+		"type": "Microsoft.Network/publicIPAddresses",
+		"name": "[concat(variables('publicIPAddressName'), copyIndex())]",
+		"location": "[variables('location')]",
+		"copy": {
+			"name": "pubIpLoop",
+			"count": "[parameters('slaveCount')]"
+		},
+		"properties": {
+			"publicIPAllocationMethod": "[variables('publicIPAddressType')]",
+			"dnsSettings": {
+				"domainNameLabel": "[concat(parameters('dnsNameForPublicIP'), copyIndex())]"
+			}
+		}
+	},
+        {
+              "apiVersion": "2015-05-01-preview",
+              "type": "Microsoft.Network/virtualNetworks",
+              "name": "[variables('virtualNetworkName')]",
+              "location": "[variables('location')]",
+              "dependsOn": [
+			"[variables('nsgID')]"
+		],
+              "properties": {
+                "addressSpace": {
+                  "addressPrefixes": [
+                    "[variables('addressPrefix')]"
+                  ]
+                },
+                "subnets": [
+                  {
+                    "name": "[variables('subnetName')]",
+                    "properties" : {
+                      "addressPrefix": "[variables('subnetPrefix')]",
+		      "networkSecurityGroup" : {
+		              "id": "[variables('nsgID')]"
+			}
+                    }
+                  }
+                ]
+               }
+        },
+	{
+		"apiVersion": "2015-05-01-preview",
+		"type": "Microsoft.Network/networkSecurityGroups",
+		"name": "[variables('nsgName')]",
+		"location": "[variables('location')]",
+		"properties": {
+			"securityRules": [
+				{
+					"name": "swarmManagementPort",
+					"properties": {
+						"description": "Docker swarm endpoint",
+						"protocol": "Tcp",
+						"sourcePortRange": "*",
+						"destinationPortRange": "5555",
```

Comment:
use 2376, it's an official port number. 5555 probably belongs to something else.